client/wc-api/settings/selectors.js

```diff
@@ -1,14 +1,35 @@
 /** @format */
 
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import { DEFAULT_REQUIREMENT } from '../constants';
 
 const getSettings = ( getResource, requireResource ) => ( requirement = DEFAULT_REQUIREMENT ) => {
-	return requireResource( requirement, 'settings' ).data;
+	return requireResource( requirement, 'settings' ).data || {};
+};
+
+const getSettingsError = getResource => () => {
```

Comment:
Should we add some tests for these new methods here?

Reply:
I don't think we have tests around any of the functions in `wc-api` as of right now.  Certainly not a bad idea though- do we want to implement tests for all of these?

I can add some in pending @coderkevin's response as to whether or not we continue using `wc-api` here or create a separate store.